### DISCUSSION TITLE
pool and client

### DISCUSSION THREAD

**Discussion by skelawsky** [porsager/postgres#693] [Q&A] [answered]
Hi, in the past I used node-postgres and there I had to choose between client and pool. In postgresjs there is no such a thing, am I right or am I missing something? 

I am little confused tbh. Is establishing connection (like below) only needed and it could be used either for transactions and normal queries?

```
const sql = postgres('postgres://username:password@host:port/database', {
  host                 : '',            // Postgres ip address[s] or domain name[s]
  port                 : 5432,          // Postgres server port[s]
  database             : '',            // Name of database to connect to
  username             : '',            // Username of database user
  password             : '',            // Password of database user
  ...and more
})
```

**Comment by porsager** [comment, marked as answer]
Simplicity can indeed be confusing when coming back to it 🤣 Postgres.js has a default max of 10 connections, but if you want a single one, just set max: 1

Transactions are run using sql.begin, and will block a connection while running, but you don't need to do anything specific yourself. You can also use sql.reserve if you explicitly need to have access to a single connection from the pool for a short period, but this is not something you usually need unless you have a special case. 

Connections are handled lazily, and with the best defaults, so a connection is not made until you make a query, and they are cycled and closed if idle per the defaults. 

If you think something is lacking in the docs, you're more than welcome to create a PR where we can work on getting it added / improved 😉

**Reply by henryzhang03** [reply]
hey @porsager , would you recommend still doing a connection pool on the db side? What would be the benefits/downsides? I'm struggling a bit to understand if postgresjs has inbuilt pooling, if db side pooling is useful at all.
